source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -31,6 +30,16 @@ DnsCacheImpl::DnsCacheImpl(
               config, refresh_interval_.count(), random)),
       host_ttl_(PROTOBUF_GET_MS_OR_DEFAULT(config, host_ttl, 300000)),
       max_hosts_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, max_hosts, 1024)) {
+  auto dns_resolver_options =
+      envoy::config::core::v3::DnsResolverOptions(config.dns_resolver_options());
+  // Field bool `use_tcp_for_dns_lookups` will be deprecated in future. To keep supporting earlier
+  // implementation of control plane APIs only accept this value if `use_tcp_for_dns_lookups`
+  // is not set via dns_resolver_options but is set `true` in the field bool
+  // `use_tcp_for_dns_lookups`.
+  if (config.use_tcp_for_dns_lookups() && !dns_resolver_options.has_use_tcp_for_dns_lookups()) {
+    dns_resolver_options.mutable_use_tcp_for_dns_lookups()->set_value(true);
+  }
```

Comment:
Please doublecheck that this is doing what we expect?

Reply:
Same explanation as above. We want user to be able to use "TCP over UDP" by setting the deprecated field _config.use_tcp_for_dns_lookups_ to true. Before we check _config.use_tcp_for_dns_lookups_ we make sure _**config. dns_resolver_options.use_tcp_for_dns_lookups**_ is not already programmed (be it true or false).